packages/next/src/server/base-server.ts

```diff
@@ -2091,6 +2091,19 @@ export default abstract class Server<
 
     const prerenderManifest = this.getPrerenderManifest()
 
+    // When bots crawl a site, they may render the page after awhile (e.g. re-sync),
+    // and the sub-assets may not be available then. In this case, the link to
+    // static assets could be not found, and return a 404 HTML. This behavior can
+    // bait the bots as if they found 404 pages. Therefore, we return a plain text
+    // "Not Found" for not found static assets. Use "includes()" to avoid issues
+    // with basePath and assetPrefix.
+    if (is404Page && req.url.includes('/_next/static/')) {
```

Comment:
There's a parsed url above with `new URL`, and we know assetPrefix, we can do a more accurate check with `startsWith`